# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/MacroAssembler.cpp
# Commit: 5737845e8af2
# Full Hash: 5737845e8af228fab6e5e622863663faa50f93e8
# Author: Dmitry Bezhetskov <dbezhetskov@igalia.com>
# Date: 2020-10-30 16:09:44
# Regressor Bug: 1639153
# File Overlap Count: 1
# Description:
#   Bug 1639153 - Part 5: Remove remaining uses of Frame::tls. r=lth
#   
#   
#   Here we remove remaining uses of Frame::tls. There are many places where
#   we use it: in debug frames, in profiling frames, in jit activation, etc.
# ==============================================================================

diff -r 5dfb9fc15fba -r 5737845e8af2 js/src/jit/MacroAssembler.cpp
--- a/js/src/jit/MacroAssembler.cpp	Fri Oct 23 18:17:37 2020 +0000
+++ b/js/src/jit/MacroAssembler.cpp	Fri Oct 23 18:17:38 2020 +0000
@@ -4662,16 +4662,16 @@
 }  // namespace jit
 
 namespace wasm {
-TlsData* ExtractCallerTlsFromFrameWithTls(Frame* fp) {
-  return *reinterpret_cast<TlsData**>(reinterpret_cast<uint8_t*>(fp) +
-                                      sizeof(Frame) + ShadowStackSpace +
-                                      FrameWithTls::callerTLSOffset());
+const TlsData* ExtractCallerTlsFromFrameWithTls(const Frame* fp) {
+  return *reinterpret_cast<TlsData* const*>(
+      reinterpret_cast<const uint8_t*>(fp) + sizeof(Frame) + ShadowStackSpace +
+      FrameWithTls::callerTLSOffset());
 }
 
-TlsData* ExtractCalleeTlsFromFrameWithTls(Frame* fp) {
-  return *reinterpret_cast<TlsData**>(reinterpret_cast<uint8_t*>(fp) +
-                                      sizeof(Frame) + ShadowStackSpace +
-                                      FrameWithTls::calleeTLSOffset());
+const TlsData* ExtractCalleeTlsFromFrameWithTls(const Frame* fp) {
+  return *reinterpret_cast<TlsData* const*>(
+      reinterpret_cast<const uint8_t*>(fp) + sizeof(Frame) + ShadowStackSpace +
+      FrameWithTls::calleeTLSOffset());
 }
 }  // namespace wasm
 